vumi/components/tests/test_message_store.py

```diff
@@ -112,6 +112,27 @@ def create_outbound_messages(self, batch_id, count, start_timestamp=None,
             messages.append(msg)
         returnValue(messages)
 
+    def _create_event(self, event_type, timestamp):
+        assert event_type in ('ack', 'nack', 'delivery_report')
+        maker = getattr(self.msg_helper, 'make_%s' % (event_type,))
+        return maker(timestamp=timestamp)
```

Comment:
Hrm. Maybe dict-based dispatch (instead of the assert and `getattr`), then? I'm not enthusiastic about using `getattr` for this stuff.